File: HSAPI/src/main/kotlin/com/hobbystock/types/ProjectItem.kt
package com.hobbystock.types

import java.util.UUID

data class ProjectItem(
        val id: UUID,
        val projectId: UUID,
        val itemId: UUID,
        val quantityUsed: Float?,
        val createdAt: String
)

data class ProjectItemInput(val projectId: String, val itemId: String, val quantityUsed: Float?)

data class ProjectItemMutationResult(
        val success: Boolean,
        val message: String?,
        val projectItem: ProjectItem?
)

File: app/src/types/ProjectItem.ts
import { Item } from "./Item";

export type ProjectItem = {
  id: number;
  projectId: number;
  itemId: number;
  quantityUsed: number;
  createdAt: string;
  item?: Item;
};

export type ProjectItemInput = {
  projectId: number;
  itemId: number;
  quantityUsed: number;
};
